addons/html_builder/static/src/website_builder/plugins/options/google_maps_option/google_maps_option_plugin.js

```diff
@@ -0,0 +1,355 @@
+        // again afterwards, it will simply recreate it.
+        for (const container of document.body.querySelectorAll('.pac-container')) {
+            container.remove();
+        }
+    }
+
+    getActions() {
+        return {
+            resetMapColor: {
+                apply: ({ editingElement }) => {
+                    editingElement.dataset.mapColor = "";
+                },
+            },
+            showDescription: {
+                isApplied: ({ editingElement }) => !!editingElement.querySelector(".description"),
+                apply: ({ editingElement }) => {
+                    editingElement.append(renderToElement("html_builder.GoogleMapsDescription"));
+                },
+                clean: ({ editingElement }) => {
+                    editingElement.querySelector(".description").remove();
+                },
+            },
+        };
+    }
+
+    /**
+     * Get the stored API key if any (or open a dialog to ask the user for one),
+     * load and configure the Google Maps API, then use it to set the pin's
+     * initial address.
+     *
+     * @param {Element} editingElement
+     * @param {boolean} [forceReconfigure=false]
+     * @returns {Promise<void>}
+     */
+    async loadGoogleMaps(editingElement, forceReconfigure = false) {
+        /** @type {string  |undefined} */
+        let apiKey = await this.googleMapsService.getGMapsAPIKey();
+        const didReconfigure = await this.configureGMapsAPI({ apiKey, force: forceReconfigure });
+        apiKey = await this.loadGoogleMapsAPIFromService(didReconfigure);
+        if (apiKey) {
+            await this.onGoogleMapsLoaded(editingElement);
+        }
+    }
+
+    /**
+     * Load the Google Maps API from the Google Map Service.
+     * This method is set apart so it can be overridden for testing.
+     *
+     * @param {boolean} [shouldRefetch]
+     * @returns {Promise<string|undefined>} A promise that resolves to an API
+     *                                      key if found.
+     */
+    async loadGoogleMapsAPIFromService(shouldRefetch) {
+        return this.googleMapsService.loadGMapsAPI(true, shouldRefetch);
+    }
+
+    /**
+     * Try to fail early in the case of a misconfiguration by doing a nearby
+     * search on the coordinates set on the editing element. If it works, set
+     * the result's address as the pin's initial address. Otherwise, remove the
+     * snippet (unless the error is being handled).
+     *
+     * @param {Element} editingElement
+     * @returns {Promise<void>}
+     */
+    async onGoogleMapsLoaded(editingElement) {
+        if (window.google) { // This should always be true, except for in tests.
+            this.mapsAPI = google.maps;
+            this.placesAPI = google.maps.places;
+        }
+        const place = await this.nearbySearch(editingElement, editingElement.dataset.mapGps);
+        if (place?.formatted_address) {
+            editingElement.dataset.pinAddress = place.formatted_address;
```

Comment:
The mutation could be problematic with the history (same with mother other mutations in this PR)
Do we want a new step to be created?
If so, you should use the mutex (OperationPlugin.next)
If we don't want mutation and the attribute to be saved, add it in `system_attributes`.
Do we want it in the dom without a step? In that we need to think more about that case.

Reply:
I don't think we need a step for this. It's just setting a technical attribute for use by the template of the option. When opening the option we need to load the API anyway so this would be done again.

Reply:
Discussed IRL.